### ISSUE TITLE
Dry::Validation::MissingMessageError after update dry stack

### ISSUE THREAD

**Issue by bestwebua** [dry-rb/dry-validation#717]
## Describe the bug

After updating dry stack I got unexpected i18n translation errors.

## To Reproduce

Dry deps:

```
dry-auto_inject (0.9.0)
  dry-container (>= 0.3.4)
dry-configurable (0.16.1)
  dry-core (~> 0.6)
  zeitwerk (~> 2.6)
dry-container (0.11.0)
  concurrent-ruby (~> 1.0)
dry-core (0.9.0)
  concurrent-ruby (~> 1.0)
  zeitwerk (~> 2.6)
dry-equalizer (0.3.0)
dry-events (0.4.0)
  concurrent-ruby (~> 1.0)
  dry-core (~> 0.9, >= 0.9)
dry-inflector (0.3.0)
dry-initializer (3.1.1)
dry-logic (1.3.0)
  concurrent-ruby (~> 1.0)
  dry-core (~> 0.9, >= 0.9)
  zeitwerk (~> 2.6)
dry-matcher (0.9.0)
  dry-core (~> 0.4, >= 0.4.8)
dry-monads (1.4.0)
  concurrent-ruby (~> 1.0)
  dry-core (~> 0.7)
dry-schema (1.11.1)
  concurrent-ruby (~> 1.0)
  dry-configurable (~> 0.16, >= 0.16)
  dry-core (~> 0.9, >= 0.9)
  dry-initializer (~> 3.0)
  dry-logic (~> 1.3)
  dry-types (~> 1.6)
  zeitwerk (~> 2.6)
dry-struct (1.5.0)
  dry-core (~> 0.9, >= 0.9)
  dry-types (~> 1.6)
  ice_nine (~> 0.11)
  zeitwerk (~> 2.6)
dry-system (0.27.1)
  dry-auto_inject (>= 0.4.0)
  dry-configurable (~> 0.16, >= 0.16.0)
  dry-container (~> 0.10, >= 0.10.0)
  dry-core (~> 0.9, >= 0.9.0)
  dry-inflector (~> 0.1, >= 0.1.2)
  zeitwerk (~> 2.6)
dry-transaction (0.13.3)
  dry-container (>= 0.2.8)
  dry-events (>= 0.1.0)
  dry-matcher (>= 0.7.0)
  dry-monads (>= 0.4.0)
dry-transformer (0.1.1)
dry-types (1.6.1)
  concurrent-ruby (~> 1.0)
  dry-container (~> 0.3)
  dry-core (~> 0.9, >= 0.9)
  dry-inflector (~> 0.1, >= 0.1.2)
  dry-logic (~> 1.3, >= 1.3)
  zeitwerk (~> 2.6)
dry-validation (1.9.0)
  concurrent-ruby (~> 1.0)
  dry-container (~> 0.7, >= 0.7.1)
  dry-core (~> 0.9, >= 0.9)
  dry-initializer (~> 3.0)
  dry-schema (~> 1.11, >= 1.11.0)
  zeitwerk (~> 2.6)
```

Source code:

```ruby
System::Container.register_provider(:core) do
  prepare do
    require 'i18n'
    require 'dry-auto_inject'
    require 'dry-container'
    require 'dry-validation'
    require 'dry-transaction'
    require 'dry-transformer'
  end

  start do
    I18n.load_path << Dir["#{File.expand_path('./system/locales')}/*.yml"]
    Dry::Validation.load_extensions(:monads)
  end
end

class Contract < Dry::Validation::Contract
  config.messages.default_locale = :en
  config.messages.top_namespace = :contract
  config.messages.backend = :i18n

  register_macro(:email_format) do
    key.failure(:email_format) unless EMAIL_PATTERN.match?(value)
  end

  params do
    required(:email).filled(:string)
  end

  rule(:email).validate(:email_format)
end
```

Translation and output:

```
# system/locales/en.yml
en:
  contract:
    errors:
      email_format: not a valid email format

# output
Dry::Validation::MissingMessageError:
       Message template for :email_format under "email" was not found
```

## Expected behavior

Translations should be found.

## My environment

- Affects my production application: **YES**
- Ruby version: `ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [x86_64-darwin20]`

**Comment by solnic** [comment]
Gaah super sorry about this :( I just fixed it in dry-schema and it'll be released in dry-schema 1.11.2

**Comment by bestwebua** [comment]
@solnic Peter, thanks for your huge work! Now it works brilliant! ❤️
